core-data/selectors.js

```diff
@@ -145,3 +145,28 @@ export const getEntityRecords = createSelector(
 export function getThemeSupports( state ) {
 	return state.themeSupports;
 }
+
+/**
+ * Returns the embed preview for the given URL.
+ *
+ * @param {Object} state    Data state.
+ * @param {string} url      Embedded URL.
+ *
+ * @return {Mixed} Undefined if the preview has not been fetched, false if the URL cannot be embedded, array of embed preview data if the preview has been fetched.
+ */
+export function getPreview( state, url ) {
```

Comment:
To disambiguate, could we name the selector `getEmbedPreview`? I might imagine this could be mistaken for e.g. a post preview.